### PR TITLE
Rework port dependency 

### PR DESCRIPTION
**Pull request by vasant17**
**What I did**
Removed code to track port dependency in orchagent. Using SAI REDIS return code to see if port deletion failed due to port dependency

**Why I did it**
So, that we can rely on SAI REDIS code for dependency check instead of duplicating the effort in orchagent.

**How I verified it**
Ran DPB command via CLI, and verified in syslog that dependency is really caught and correct log messages are printed. Also verified that breakout itself goes through successfully when dependencies are removed.

**Details if related**

### CODE REVIEW

File: orchagent/portsorch.cpp [hunk 1]
```diff
@@ -2304,27 +2308,21 @@ void PortsOrch::doPortTask(Consumer &consumer)
...
-            }
-
-            deinitport(alias, port_id);
-
-            SWSS_LOG_NOTICE("Removing hostif %lx for Port %s", hif_id, alias.c_str());
-            sai_status_t status = sai_hostif_api->remove_hostif(hif_id);
-            if (status != SAI_STATUS_SUCCESS)
-            {
-                throw runtime_error("Remove hostif for the port failed");
+            if (m_portList[alias].m_init) {
```
Comment (zhenggen-xu) [inline orchagent/portsorch.cpp]: Let's make the format consistent.  move the { to below line.

File: orchagent/portsorch.cpp [hunk 2]
```diff
@@ -2304,27 +2308,21 @@ void PortsOrch::doPortTask(Consumer &consumer)
...
+                {
+                    throw runtime_error("Remove hostif for the port failed");
+                }
+                m_portList[alias].m_init = false;
             }
 
             if (!removePort(port_id))
             {
-                throw runtime_error("Delete port failed");
+                it++;
```
Comment (zhenggen-xu) [inline orchagent/portsorch.cpp:2337]: code below removePort() could be called many times.  Do you see issues?
```
    Port p;
    if (getPort(port_id, p))
    {
        PortUpdate update = {p, false };
        notify(SUBJECT_TYPE_PORT_CHANGE, static_cast<void *>(&update));
    }
```

File: orchagent/portsorch.cpp [hunk 3]
```diff
@@ -2303,28 +2307,28 @@ void PortsOrch::doPortTask(Consumer &consumer)
...
             auto port_id = m_portList[alias].m_port_id;
             auto hif_id = m_portList[alias].m_hif_id;
+            auto bridge_port_oid = m_portList[alias].m_bridge_port_id;
 
-            if (m_portList[alias].has_dependency())
-            {
-                // Port has one or more dependencies, cannot remove
-                SWSS_LOG_WARN("Please remove port dependenc(y/ies):%s",
-                               m_portList[alias].print_dependency().c_str());
+            if (bridge_port_oid != SAI_NULL_OBJECT_ID) {
```
Comment (zhenggen-xu) [inline orchagent/portsorch.cpp]: Can you put a comment here for what cases we are deferring here?   Also, let's log this event to syslog.

File: orchagent/portsorch.cpp (hunk 2 above)
Comment (vasant17) [reply orchagent/portsorch.cpp:2337]: No

Comment (zhenggen-xu) [reply orchagent/portsorch.cpp:2337]: To make it symmetric with addport() and void any re-entering issue for notify() for different observers. I would suggest to move this part of code to outside, and we should not call it again and again during the retry.

Comment (vasant17) [reply orchagent/portsorch.cpp:2337]: Agree. Moved it out of the function and inside the init block.

**Review by zhenggen-xu — APPROVED** [review]

**Pull request merged**